configure.ac

```diff
@@ -136,43 +136,58 @@ AC_CHECK_HEADER(valgrind/memcheck.h, [],
 fi
 
 AC_CACHE_CHECK(whether ld accepts --version-script, ac_cv_version_script,
-    if test -n "`$LD --help < /dev/null 2>/dev/null | grep version-script`"; then
+    [if test -n "`$LD --help < /dev/null 2>/dev/null | grep version-script`"; then
         ac_cv_version_script=yes
     else
         ac_cv_version_script=no
-    fi)
+    fi])
 
 AM_CONDITIONAL(HAVE_LD_VERSION_SCRIPT, test "$ac_cv_version_script" = "yes")
 
 dnl Disable symbol versioning with icc + ipo, with it ipo is disabled by icc.
 dnl The gcc equivalent ipo (-fwhole-program) seems to work fine.
 AS_IF([case "$CFLAGS" in
-         *-ipo*) true ;;
-         *)      false ;;
-        esac],
-      [AC_MSG_NOTICE([disabling symbol versioning support with -ipo CFLAG])],
-[
-
-AC_CACHE_CHECK(for .symver assembler support, ac_cv_asm_symver_support,
-	[AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[]],
-		[[asm("symbol:\n.symver symbol, api@ABI\n");]])],
-		[ac_cv_asm_symver_support=yes],
-		[ac_cv_asm_symver_support=no])])
-
-]) dnl AS_IF
-
-if test x$ac_cv_asm_symver_support = xyes; then
-	AC_DEFINE([HAVE_SYMVER_SUPPORT], 1, [assembler has .symver support])
-fi
+	*-ipo*) true ;;
+	*)      false ;;
+	esac],
+	[
+		AC_MSG_NOTICE([disabling symbol versioning support with -ipo CFLAG])
+		ac_icc_hack=0
+	],
+	[ac_icc_hack=1])
+
+AC_MSG_CHECKING(for .symver assembler support)
+AC_TRY_LINK([],
+	[__asm__(".symver main_, main@@ABIVER_1.0");],
+	[
+		AC_MSG_RESULT(yes)
+		ac_cv_asm_symver_support=$ac_icc_hack
+	],
+	[
+		AC_MSG_RESULT(no)
+		ac_cv_asm_symver_support=0
+	])
+
 
-AC_LINK_IFELSE(
-	[AC_LANG_SOURCE[
+AC_DEFINE_UNQUOTED([HAVE_SYMVER_SUPPORT], [$ac_cv_asm_symver_support],
+	  	   [Define to 1 if compiler/linker support symbol versioning.])
+
+AC_MSG_CHECKING(for __alias__ attribute support)
+AC_TRY_LINK(
+	[
 		int foo(int arg);
 		int foo(int arg) { return arg + 3; };
 		int foo2(int arg) __attribute__ (( __alias__("foo")));
-	]],
-	[ac_cv_prog_cc_alias_symbols=1],
-	[ac_cv_prog_cc_alias_symbols=0])
+	],
+	[ /* empty main */ ],
+	[
+		AC_MSG_RESULT(yes)
+		ac_cv_prog_cc_alias_symbols=1
```

Comment:
Isn't ac_cv is supposed to be reserved for the prefix for autoconf cache variables?